pymbolic/mapper/__init__.py

```diff
@@ -704,7 +989,7 @@ class CachedIdentityMapper(CachedMapper, IdentityMapper):
 
 # {{{ walk mapper
 
-class WalkMapper(RecursiveMapper):
+class WalkMapper(Mapper):
```

Comment:
Here is a diff for the rest of WalkMapper: https://gist.github.com/matthiasdiener/fa0bacd4b54cffa49218faaf4868840a

Reply:
Thanks for doing this. I won't take it though. All the type-ignores are unnecessary, and I would very much like to avoid `Any` types if at all possible.